core/runtime/wavm/module_instance.cpp

```diff
@@ -42,67 +42,68 @@ namespace kagome::runtime::wavm {
-    invokeArgs.emplace_back(static_cast<WAVM::U32>(args.size));
-    WAVM_ASSERT(function)
+    auto res = [this, name, args]() -> outcome::result<PtrSize> {
+      WAVM::Runtime::GCPointer<WAVM::Runtime::Context> context =
+          WAVM::Runtime::createContext(compartment_->getCompartment());
+      WAVM::Runtime::Function *function = WAVM::Runtime::asFunctionNullable(
+          WAVM::Runtime::getInstanceExport(instance_, name.data()));
+      std::vector<WAVM::IR::Value> invokeArgs;
+      if (!function) {
+        SL_DEBUG(logger_, "The requested function {} not found", name);
+        return Error::FUNC_NOT_FOUND;
+      }
+      const WAVM::IR::FunctionType functionType =
+          WAVM::Runtime::getFunctionType(function);
+      if (functionType.params().size() != 2) {  // address and size
+        SL_DEBUG(
+            logger_,
+            "The provided function argument count should equal to 2, got {} "
+            "instead",
+            functionType.params().size());
+        return Error::WRONG_ARG_COUNT;
+      }
+      invokeArgs.emplace_back(static_cast<WAVM::U32>(args.ptr));
+      invokeArgs.emplace_back(static_cast<WAVM::U32>(args.size));
+      WAVM_ASSERT(function)
 
-    std::vector<WAVM::IR::ValueType> invokeArgTypes;
-    std::vector<WAVM::IR::UntaggedValue> untaggedInvokeArgs;
-    for (const WAVM::IR::Value &arg : invokeArgs) {
-      invokeArgTypes.push_back(arg.type);
-      untaggedInvokeArgs.push_back(arg);
-    }
-    // Infer the expected type of the function from the number and type of the
-    // invoke arguments and the function's actual result types.
-    const WAVM::IR::FunctionType invokeSig(
-        WAVM::Runtime::getFunctionType(function).results(),
-        WAVM::IR::TypeTuple(invokeArgTypes));
-    // Allocate an array to receive the invoke results.
-    std::vector<WAVM::IR::UntaggedValue> untaggedInvokeResults;
-    untaggedInvokeResults.resize(invokeSig.results().size());
-    try {
-      WAVM::Runtime::unwindSignalsAsExceptions([&context,
-                                                &function,
-                                                &invokeSig,
-                                                &untaggedInvokeArgs,
-                                                &untaggedInvokeResults] {
-        WAVM::Runtime::invokeFunction(context,
-                                      function,
-                                      invokeSig,
-                                      untaggedInvokeArgs.data(),
-                                      untaggedInvokeResults.data());
-      });
-      return PtrSize{untaggedInvokeResults[0].u64};
-    } catch (WAVM::Runtime::Exception *e) {
-      const auto desc = WAVM::Runtime::describeException(e);
-      logger_->debug(desc);
-      WAVM::Runtime::destroyException(e);
-      return Error::EXECUTION_ERROR;
-    }
+      std::vector<WAVM::IR::ValueType> invokeArgTypes;
+      std::vector<WAVM::IR::UntaggedValue> untaggedInvokeArgs;
```

Comment:
we could resize these vectors before for in the next line

Reply:
Turned out they are not needed at all, as there are always two 32 byte arguments.